### PR TITLE
website/integrations: add three more policy-expressions to discord-docs

### PR DESCRIPTION
**Pull request by Keyinator**
### New Features

- Add two policy-examples for syncing discord roles to authentik groups during enrollment or authentication

- Add policy to store oauth-info and create an authentik-avatar-attribute during enrollment or authentication

## Checklist

-   [X] The documentation has been updated
-   [ ] The documentation has been formatted (`make website`)

### CODE REVIEW

**Comment by tanberry** [comment]
Hi @Keyinator thanks for this PR. It is marked as a draft... are you ready for it to be reviewed and hopefully merged?

**Comment by BeryJu** [comment]
Hey @Keyinator thanks for the PR, do you want to make any other changes before we review or should we change a couple of final things ourselves and merge it?

**Comment by Keyinator** [comment]
> Hey @Keyinator thanks for the PR, do you want to make any other changes before we review or should we change a couple of final things ourselves and merge it?

Sorry for the late reply, I did not notice that the pr was awaiting an answer.

I will have another look at it over the weekend and see if there's any more details that can be improved.

File: website/integrations/sources/discord/index.md [hunk 1]
```diff
@@ -154,3 +154,205 @@ return user_matched
...
+Now bind this policy to the chosen authentication flows for the Discord OAuth source.
+
+### Store oauth info in attribute and create avatar attribute from discord avatar
+
+:::info
+Ensure that the Discord OAuth source in 'Federation & Social login' has the additional `guilds.members.read` scopes added under the 'Protocol settings'.
+:::
+
+:::info
+In order to use the created attribute in authentik you will have to set authentik configuration arguments found at: https://goauthentik.io/docs/installation/configuration#authentik_avatars
```
Comment (Codixer) [inline website/integrations/sources/discord/index.md:296]: They might have moved the location of the avatars here:
https://docs.goauthentik.io/docs/core/settings#avatars

File: website/integrations/sources/discord/index.md [hunk 2]
```diff
@@ -154,3 +154,205 @@ return user_matched
...
+# enable developer mode.
+# Right-click on the server/guild title and select "Copy ID" to get the guild ID.
+# Right-click on the server/guild title and select server settings > roles, right click on the role and click
+# "Copy ID" to get the role ID.
+
+from authentik.core.models import Group
+
+GUILD_ID = "<YOUR GUILD ID>"
+MAPPED_ROLES = {
+    "<Discord Role Id>": Group.objects.get_or_create(name="<Authentik Role Name>")[0],
```
Comment (Codixer) [inline website/integrations/sources/discord/index.md:237]: Out of curiosity, what happens when I have several roles I wanna apply to a person? Like they have "Rank 1" and "Rank 2", and I wanna apply both Group 1 and Group 2 to them. Would I have to apply 2 mapped roles?

Comment (Keyinator) [reply website/integrations/sources/discord/index.md:237]: I'll add a second entry so it's clearer but that's a list so you would just add another entry with the values for your other rank. All ranks are checked whether they apply and all qualifying will be applied

File: website/integrations/sources/discord/index.md (hunk 1 above)
Comment (Keyinator) [reply website/integrations/sources/discord/index.md:296]: You're right. I'll fix that :)

File: website/integrations/sources/discord/index.md (hunk 2 above)
Comment (Codixer) [reply website/integrations/sources/discord/index.md:237]: Don't forget it, the team is waiting on your approval. Whahahah

File: website/integrations/sources/discord/index.md [hunk 3]
```diff
@@ -154,3 +154,205 @@ return user_matched
 ```
 
 Now bind this policy to the chosen enrollment and authentication flows for the Discord OAuth source.
+
+### Syncing discord roles to authentik groups
```
Comment (tanberry) [inline website/integrations/sources/discord/index.md:158]:
```suggestion
### Syncing Discord roles to authentik groups
```

File: website/integrations/sources/discord/index.md [hunk 4]
```diff
@@ -154,3 +154,205 @@ return user_matched
...
+
+:::info
+Ensure that the Discord OAuth source in 'Federation & Social login' has the additional `guilds.members.read` scopes added under the 'Protocol settings'.
+:::
+
+Create a new 'Expression Policy' with the content below, adjusting the variables where required:
```
Comment (tanberry) [inline website/integrations/sources/discord/index.md:164]:
```suggestion
Create a new 'Expression Policy' with the content below, adjusting the variables where required.
```

Comment (tanberry) [reply website/integrations/sources/discord/index.md:164]: @Keyinator there was a colon here... did you intend to list out specific variables? Or was it supposed to be a period?

File: website/integrations/sources/discord/index.md [hunk 5]
```diff
@@ -154,3 +154,205 @@ return user_matched
...
+request.user.ak_groups.set(new_groups)
+request.user.save()
+
+return True
+
+```
+
+Now bind this policy to the chosen authentication flows for the Discord OAuth source.
+
+### Store oauth info in attribute and create avatar attribute from discord avatar
```
Comment (tanberry) [inline website/integrations/sources/discord/index.md:289]:
```suggestion
### Store OAuth info in attribute and create avatar attribute from Discord avatar
```

**Review by tanberry — CHANGES_REQUESTED** [review]
Thanks @Keyinator for this contribution! A couple of nit-picks about capitalization, and one question, then let's get this merged!

**Comment by Keyinator** [comment]
All done @tanberry

**Review by tanberry — APPROVED** [review]

**Comment by tanberry** [comment]
@Keyinator this PR apparently has merge conflicts... do you want to resolve the and push again, or I can... clone your branch and do it...? I think, lol.

**Comment by Keyinator** [comment]
> @Keyinator this PR apparently has merge conflicts... do you want to resolve the and push again, or I can... clone your branch and do it...? I think, lol.

Should be fixed now

**Comment by tanberry** [comment]
Thank you so much for this contribution and for your perseverance!

**Pull request merged**